src/main/java/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,134 @@
+import bookstore.model.User;
+import bookstore.repository.OrderItemRepository;
+import bookstore.repository.OrderRepository;
+import bookstore.repository.ShoppingCartRepository;
+import bookstore.service.OrderService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderResponseDto createOrder(User user, OrderRequestDto orderRequestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(user.getId())
+                .orElseThrow(() -> new EntityNotFoundException("There is no user with the id: "
+                        + user.getId()));
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+        if (cartItems.isEmpty()) {
+            throw new OrderException("There is no order");
+        }
+        Order order = new Order();
+        order.setUser(user);
+        order.setOrderDate(LocalDateTime.now());
+        order.setStatus(Order.Status.CREATED);
+        order.setShippingAddress(orderRequestDto.shippingAddress());
+        order.setOrderItems(createOrderItems(cartItems, order));
+        order.setTotal(countTotalSum(order.getOrderItems()));
+        shoppingCart.getCartItems().clear();
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderResponseDto> findAllOrders(Long userId, Pageable pageable) {
+        return orderRepository.findAllOrdersByUserId(userId, pageable).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public List<OrderItemResponseDto> findAllByOrderId(Long orderId,
+                                                       Pageable pageable,
+                                                       Long userId) {
+        Order order = checkIfOrderExistsById(orderId);
+        checkIfOrderBelongsToUser(order.getUser().getId(), userId, orderId);
+        return orderItemRepository.findAllByOrderId(orderId, pageable)
+                .stream()
+                .map(orderItemMapper:: toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemResponseDto findOrderItemByIdInOrder(Long orderId,
+                                                         Long orderItemId, Long userId) {
+        Order order = checkIfOrderExistsById(orderId);
```

Comment:
Just FYI, you can use one query to DB to find the orderItem

```
OrderItem item = itemRepository.findByIdAndOrderIdAndUserId(orderItemId, orderId, userId)
                .orElseThrow(() -> new EntityNotFoundException(
                        String.format("Order item: %d not found in order: %d for user: %d",
                                orderItemId, orderId, userId)
                ));
```
                
Where in OrderRepository 
```
@Query(value = """
            SELECT oi FROM OrderItem oi
            JOIN FETCH oi.order o
            WHERE oi.id = :orderItemId AND o.id = :orderId AND o.user.id = :userId""")
    Optional<OrderItem> findByIdAndOrderIdAndUserId(@Param("orderItemId") Long orderItemId,
                                                    @Param("orderId") Long orderId,
                                                    @Param("userId") Long userId);
```